conda/activate.py

```diff
@@ -188,18 +188,17 @@ def reactivate(self):
             self._yield_commands(self.build_reactivate()), self.tempfile_extension
         )
 
-    def hook(self, auto_activate_base: bool | None = None) -> str:
+    @deprecated.argument("25.5", "26.1", "auto_activate_base", rename="auto_activate")
+    def hook(self, auto_activate: bool | None = None) -> str:
         builder: list[str] = []
         if preamble := self._hook_preamble():
             builder.append(preamble)
         if self.hook_source_path:
             builder.append(self.hook_source_path.read_text())
-        if (
-            auto_activate_base is None
-            and context.auto_activate_base
-            or auto_activate_base
-        ):
-            builder.append("conda activate base\n")
+        if auto_activate is None and context.auto_activate or auto_activate:
+            builder.append(
+                f"conda activate '{context.default_activation_env or ROOT_ENV_NAME}'\n"
```

Comment:
Since we do this in several places are we not better off doing this in a property in the `context` object?

```python
class Context(...):
     _default_activation_env = ParameterLoader(PrimitiveParameter(ROOT_ENV_NAME), aliases=("default_activation_env",)

     @property
     def default_activation_env(self) -> str:
         return self._default_activation_env or ROOT_ENV_NAME
```

Reply:
Oh, nice, yes.